hedera-node/hedera-app/src/test/java/com/hedera/node/app/grpc/HelidonGrpcServerManagerTest.java

```diff
@@ -0,0 +1,333 @@
+            // we may proceed with the rest of the test
+            while (true) {
+                final var logs = String.join("\n", logCaptor.infoLogs());
+                if (logs.contains("Still trying to start server... 9 tries remaining")) {
+                    break;
+                }
+                TimeUnit.MILLISECONDS.sleep(10);
+            }
+
+            // Stop the server that was using the port
+            serverSocket.close();
+
+            // Wait for the server to start
+            while (true) {
+                final var logs = String.join("\n", logCaptor.debugLogs());
+                if (logs.contains("Helidon gRPC server listening on port")) {
+                    break;
+                }
+                TimeUnit.MILLISECONDS.sleep(10);
+            }
+
+            // FUTURE: I'm only testing the plain port and not the tls port because these tests do not yet support tls
+            // properly. But when they do, we should check the tls port too.
+            try (final var clientSocket = new Socket("localhost", subject.port())) {
+                assertThat(clientSocket.isConnected()).isTrue();
+            } catch (UnknownHostException e) {
+                fail("'localhost' was an unknown host!", e);
+            } catch (IOException e) {
+                fail("Could not connect to port " + subject.port(), e);
+            }
+        } finally {
+            subject.stop();
+        }
+    }
+
+    // Start with a port that is already in use, and observe that after N retries and X millis per retry, the server
+    // ultimately fails to start.
+    @Test
+    @Timeout(value = 50, unit = TimeUnit.SECONDS)
+    @DisplayName("Starting a server with a port already in use")
+    void portInUse() throws Exception {
+        assumePortIsFree(8588);
+        final var config = new HederaTestConfigBuilder(false)
+                .withConfigDataType(GrpcConfig.class)
+                .withConfigDataType(NettyConfig.class)
+                .withValue("grpc.port", 8588)
+                .withValue("grpc.tlsPort", 8589)
+                .withValue("netty.startRetryIntervalMs", 10)
+                .withValue("netty.startRetries", 1)
+                .getOrCreateConfig();
+        this.configProvider = () -> new VersionedConfigImpl(config, 1);
+        final var subject = new HelidonGrpcServerManager(configProvider, services);
+        try (final var serverSocket = new ServerSocket()) {
+            // Use the port
+            serverSocket.setReuseAddress(true);
+            serverSocket.bind(new InetSocketAddress(8588));
+            assertThat(serverSocket.isBound()).isTrue();
+
+            // Start the gRPC server, trying to use the same port
+            assertThatThrownBy(subject::start).isInstanceOf(RuntimeException.class);
+        } finally {
+            subject.stop();
+        }
+    }
+
+    // Missing: Not sure how to test the TLS port
```

Comment:
I think we should also add a test that starts the server, stops it, and starts it again. It happens quite easily that resources are not cleaned up properly between two starts.